pkg/inlinekeepalive/server_test.go

```diff
@@ -0,0 +1,49 @@
+package inlinekeepalive
```

Comment:
This is a tricky package to test. This covers some of it, and demonstrates the package's functionality, but if people have ideas for more tests let me know!

Reply:
I added a suite of new very thorough tests.